src/agent/at-driver.js

```diff
@@ -100,6 +109,7 @@ export class ATDriver {
    * @returns {AsyncGenerator<string>}
    */
   async *speeches() {
+    if (this.hasClosed) throw new Error('AT-Driver connection unexpectedly closed');
```

Comment:
This check is now redundant since the following line invokes the `_messages` method.